fdbserver/BlobWorker.actor.cpp

```diff
@@ -2740,11 +2749,39 @@ ACTOR Future<GranuleStartState> openGranule(Reference<BlobWorkerData> bwData, As
 	}
 }
 
+ACTOR Future<Reference<BlobConnectionProvider>> loadBStoreForTenant(Reference<BlobWorkerData> bwData,
+                                                                    KeyRange keyRange) {
+	state int retryCount = 0;
+	loop {
+		state Reference<GranuleTenantData> data = bwData->tenantData.getDataForGranule(keyRange);
+		if (data.isValid()) {
+			wait(data->bstoreLoaded.getFuture());
+			return data->bstore;
+		} else {
+			TEST(true); // bstore for unknown tenant
+			// Assume not loaded yet, just wait a bit. Could do sophisticated mechanism but will redo tenant loading to
+			// be versioned anyway
+			retryCount++;
+			TraceEvent(retryCount < 10 ? SevDebug : SevWarn, "BlobWorkerUnknownTenantForGranule", bwData->id)
+			    .detail("KeyRange", keyRange);
+			wait(delay(0.1));
+		}
+	}
+}
+
 ACTOR Future<Void> start(Reference<BlobWorkerData> bwData, GranuleRangeMetadata* meta, AssignBlobRangeRequest req) {
 	ASSERT(meta->activeMetadata.isValid());
+
+	Future<Reference<BlobConnectionProvider>> loadBStore;
+	if (SERVER_KNOBS->BG_METADATA_SOURCE != "tenant") {
```

Comment:
a nit: may be define "tenant" as a MACRO or const in common "Util" file, incase if the string gets updated in future, it won't need updating all usage as well.

Reply:
True, I am just following the pattern here that a bunch of other string knobs follow, like KMS_CONNECTOR_TYPE, DEFAULT_FDB_ROCKSDB_COLUMN_FAMILY, etc...
I agree it's probably not quite the cleanest, but on the other hand, changing a knob value isn't backwards compatible, so we already have to be careful changing these anyway.